RealmSwift/Combine.swift

```diff
@@ -880,6 +909,204 @@ public enum RealmPublishers {
                 .receive(subscriber: subscriber)
         }
     }
+    /// A helper publisher used to support `receive(on:)` on Realm publishers.
+    public class AsyncWriteRealmCombine<S: Scheduler, Output: ThreadConfined>: Publisher {
+        private class AsyncRealmWriteSubscription: Subscription {
+            func request(_ demand: Subscribers.Demand) {
+            }
+
+            func cancel() {
+            }
+
+            var combineIdentifier: CombineIdentifier {
+                CombineIdentifier(self)
+            }
+        }
+        /// :nodoc:
+        public typealias Failure = Error
+
+        private let config: Realm.Configuration
+        private let scheduler: S
+        private var subscribers = [AnySubscriber<Output, Failure>]()
+        private let block: (Realm, Output) -> Void
+        private var blocks = [(Realm, Output) -> Void]()
+        var confined: Output
+        internal init(_ scheduler: S, configuration: Realm.Configuration, confined: Output, _ block: @escaping (Realm, Output) -> Void) {
+            self.config = configuration
+            self.scheduler = scheduler
+            self.block = block
+            self.confined = confined
+            blocks.append(block)
+
+        }
+
+        /// :nodoc:
+        public func receive<Sub>(subscriber: Sub) where Sub: Subscriber, Sub.Failure == Failure, Output == Sub.Input {
+            subscribers.append(AnySubscriber(subscriber))
+            subscriber.receive(subscription: AsyncRealmWriteSubscription())
+
+            let frozenConfined1 = self.confined.freeze()
+            self.scheduler.schedule {
+                guard let thawed1 = frozenConfined1.thaw() else {
+                    return subscriber.receive(completion: .failure(NSError()))
+                }
+                let r = realm(self.config.rlmConfiguration, self.scheduler)!
+                r.beginWrite()
+                do {
+                    self.block(r, thawed1)
+                    try r.commitWrite()
+                    _ = subscriber.receive(thawed1)
+                    subscriber.receive(completion: .finished)
+                } catch {
+                    subscriber.receive(completion: .failure(error))
+                }
```

Comment:
Don't we need an autoreleasepool here to prevent the chance of the realm staying around, or has that been superseded in later versions of Swift?